src/client.rs

```diff
@@ -202,8 +195,8 @@ impl Spotify {
     /// [Reference](https://developer.spotify.com/web-api/get-track/)
     #[maybe_async]
     pub async fn track(&self, track_id: &str) -> ClientResult<FullTrack> {
-        let trid = self.get_id(Type::Track, track_id);
-        let url = format!("tracks/{}", trid);
+        let trid = Id::from_id_or_uri(Type::Track, track_id)?;
```

Comment:
Instead of this, the function should take an `Id` parameter, and we can later check that it's a `Type::Track`. That way, the user may initialize `Id` however they want. For example, if they are sure it's an URI there's no need to use `from_id_or_uri`.